packages/@ember/-internals/metal/lib/mixin.ts

```diff
@@ -575,6 +576,11 @@ export default class Mixin {
     this.ownerConstructor = undefined;
     this._without = undefined;
 
+    if (DEBUG) {
+      // Eagerly add INIT_FACTORY to avoid issues in DEBUG as a result of Object.seal(mixin)
+      this[INIT_FACTORY] = null;
+    }
+
     if (DEBUG) {
```

Comment:
Sorry @snewcomer for the run around here, would you mind merging these two `if (DEBUG) {` statements? They are back to back, and it looks a bit odd...

Reply:
I clearly don't have my thinking hat on today.  Fixed!